exp/runtime/internal/controllers/extensionconfig_controller.go

```diff
@@ -149,6 +174,29 @@ func (r *Reconciler) reconcileDelete(_ context.Context, extensionConfig *runtime
 	return ctrl.Result{}, nil
 }
 
+// secretToExtensionConfig maps a secret to ExtensionConfigs to reconcile them on updates of the secrets.
+func (r *Reconciler) secretToExtensionConfig(secret client.Object) []reconcile.Request {
```

Comment:
Just a comment. I think the list call isn't super expensive as it's just a list against the local cache. 

Nevertheless if we can skip some secrets, that's a useful performance optimization.

Reply:
Because we only act on the type `PartialObjectMetadata` here: we cannot cast to `*corev1.Secret` to access `secret.Type`.

Also `PartialObjectMetadata` does not give us more information about the secret.

The only thing I see would be accessing the managed fields and parsing them but I don't think we want to do that:
![image](https://user-images.githubusercontent.com/417870/173407255-0bd4b194-cc49-4c9c-8060-6d7440d237d0.png)

Reply:
Also: doing a get to get the real secret to check that would be:
* A: very expensive (uncached)
* B: bad for security (because we may get secrets and load them to memory which we don't need)

Reply:
no problem, this will get fixed as soon as we implement an index